tests/ethernet/test_ethernet_network.py

```diff
@@ -63,4 +64,141 @@ def test_ethernet_disconnection(connect):
+    servo, net = connect_to_slave
+    server = virtual_drive
+
+    virtual_net = EthernetNetwork()
+    protocol_contents = read_config['ethernet']
+    virtual_servo = virtual_net.connect_to_slave(
+        server.ip,
+        protocol_contents['dictionary'],
+        server.port
+    )
+    
+    virtual_response = virtual_servo.write(reg, value, subnode)
+    response = servo.write(reg, value, subnode)
+    assert response == virtual_response
+
+    response = servo.read(reg, subnode)
+    virtual_response = virtual_servo.read(reg, subnode)
+    assert response == virtual_response
+
+    old_value = virtual_servo.read(reg, subnode)
+    new_value = old_value + 1
+    virtual_servo.write(reg, new_value, subnode)
+    saved_value = virtual_servo.read(reg, subnode)
+    assert saved_value == new_value
+
+
+@pytest.mark.no_connection
+def test_load_firmware_file_not_found():
+    virtual_net = EthernetNetwork()
+    with pytest.raises(FileNotFoundError):
+        virtual_net.load_firmware("no_file")
+
+
+@pytest.mark.no_connection
+def test_load_firmware_no_connection(read_config):
+    protocol_contents = read_config['ethernet']
+    virtual_net = EthernetNetwork()
+    with pytest.raises(ILFirmwareLoadError):
+        virtual_net.load_firmware(protocol_contents["fw_file"], target="localhost", ftp_user="", ftp_pwd="")
+
+
+@pytest.mark.skip
+@pytest.mark.no_connection
+def test_load_firmware_wrong_user_pwd():
+    # TODO: implement
+    pass
+
+@pytest.mark.skip
+@pytest.mark.no_connection
+def test_load_firmware_error_during_loading():
+    # TODO: implement
+    pass
+
+
+@pytest.mark.no_connection
+def test_net_status_listener_connection(virtual_drive, read_config):
+    server = virtual_drive
+    net = EthernetNetwork()
+    protocol_contents = read_config['ethernet']
+    status_list = []
+    net.subscribe_to_status(status_list.append)
+    status_listener = NetStatusListener(net)
+
+    assert len(status_list) == 0
+
+    servo = net.connect_to_slave(
+        server.ip,
+        protocol_contents['dictionary'],
+        server.port
+    )
+
+    # Emulate a disconnection. TODO: disconnect from the virtual drive
+    status_listener._NetStatusListener__state = NET_STATE.DISCONNECTED
+    status_listener.start()
+    time.sleep(2)
+    status_listener.stop()
+
+    assert len(status_list) == 1
+    assert status_list[0] == NET_DEV_EVT.ADDED
+
+    
+@pytest.mark.skip
+@pytest.mark.no_connection
+def test_net_status_listener_disconnection():
+    # TODO: implement
+    pass
+
+
+@pytest.mark.no_connection
+def test_unsubscribe_from_status(virtual_drive, read_config):
+    server = virtual_drive
+    net = EthernetNetwork()
+    protocol_contents = read_config['ethernet']
+    status_list = []
+    net.subscribe_to_status(status_list.append)
+    net.unsubscribe_from_status(status_list.append)
+
+    status_listener = NetStatusListener(net)
```

Comment:
Same as previous. I think you can do all of these checks using the EthernetNetwork class.